services/reporting/reporting_test.go

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+package reporting
+
+import (
+	"testing"
+	"time"
+
+	"github.com/getsentry/sentry-go"
+)
+
+type TestingError struct {
+	msg string
+}
+
+func (e TestingError) Error() string {
+	return e.msg
+}
+
+type TestingTransport struct {
+	flushes chan time.Duration
+}
+
+func (t TestingTransport) Flush(timeout time.Duration) bool {
+	t.flushes <- timeout
+	return true
+}
+
+func (t TestingTransport) Configure(options sentry.ClientOptions) {}
+
+func (t TestingTransport) SendEvent(event *sentry.Event) {}
+
+type TestingEnvironment struct {
+	events  chan *sentry.Event
+	flushes chan time.Duration
+}
+
+func InitForTesting() TestingEnvironment {
+	events := make(chan *sentry.Event, 1)
+	flushes := make(chan time.Duration, 1)
+	transport := TestingTransport{flushes}
+	opts := Options()
+	opts.Transport = transport
+	opts.BeforeSend = func(event *sentry.Event, hint *sentry.EventHint) *sentry.Event {
+		events <- event
+		return event
+	}
+	Init(opts)
+
+	return TestingEnvironment{events, flushes}
+}
+
+// Reporting an error should capture its error message.
+func TestCaptureException(t *testing.T) {
+	env := InitForTesting()
+	eventID := CaptureException(TestingError{"some error"})
+	event := <-env.events
+	lastEventID := &event.EventID
+
+	if lastEventID != eventID {
```

Comment:
We use `require` and `assert` from testify for assertions, please use them, you can see examples on other tests.